### PR TITLE
HashMap implementation

### PR DESCRIPTION
**Pull request by ivantcacenco**

### CODE REVIEW

File: src/main/java/com/endava/internship/collections/StudentMap.java [hunk 1]
```diff
@@ -1,78 +1,322 @@
...
+
+    private final Pair<Student, Integer>[] buckets;
+    private static final int INITIAL_CAPACITY = 1 << 4;
+    private int size = 0;
+
+    public StudentMap() {
+        this(INITIAL_CAPACITY);
+    }
+
+    public StudentMap(int capacity) {
```
Comment (alexandrumalina) [inline src/main/java/com/endava/internship/collections/StudentMap.java]: final

File: src/main/java/com/endava/internship/collections/StudentMap.java [hunk 2]
```diff
@@ -1,78 +1,322 @@
...
+        this.buckets = new Pair[capacity];
+    }
+
+    static class Pair<K, V> {
+        private final K key;
+        private V value;
+        private final Pair<K, V> next;
+
+        public Pair(K key, V value, Pair<K, V> next) {
```
Comment (alexandrumalina) [inline src/main/java/com/endava/internship/collections/StudentMap.java]: final

**Review by alexandrumalina — APPROVED** [review]
overall lgtm
squash commits and write an exhaustive message please

File: src/main/java/com/endava/internship/collections/StudentMap.java (hunk 1 above)
Comment (ivantcacenco) [reply src/main/java/com/endava/internship/collections/StudentMap.java]: Resolved

File: src/main/java/com/endava/internship/collections/StudentMap.java (hunk 2 above)
Comment (ivantcacenco) [reply src/main/java/com/endava/internship/collections/StudentMap.java]: Resolved

**Pull request merged**